hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -364,10 +376,15 @@ private IterationResult doIteration() {
 
     // match each overUtilized node with a target
     for (DatanodeUsageInfo datanodeUsageInfo : overUtilizedNodes) {
+      if (!balancerRunning) {
+        checkIterationMoveResults();
+        return IterationResult.ITERATION_INTERRUPTED;
+      }
```

Comment:
I would recommend moving this to a separate function for checking balancer status.
Instead of calling `checkIterationMoveResults` multiple times, maybe throw InterruptedException in the new function. 
```
  doIteration {
    try {
        checkBalancerStatus();
        ...
    } catch (InterrupedException e) {
        return IterationResult.ITERATION_INTERRUPTED;
    } finally {
        checkIterationMoveResults();
    }
  }
```